public/app/features/admin/UserPermissions.tsx

```diff
@@ -10,98 +10,85 @@ interface Props {
   onGrafanaAdminChange: (isGrafanaAdmin: boolean) => void;
 }
 
-interface State {
-  isEditing: boolean;
-  currentAdminOption: string;
-}
-
 const adminOptions = [
   { label: 'Yes', value: 'YES' },
   { label: 'No', value: 'NO' },
 ];
 
-export class UserPermissions extends PureComponent<Props, State> {
-  state = {
-    isEditing: false,
-    currentAdminOption: this.props.isGrafanaAdmin ? 'YES' : 'NO',
-  };
+export function UserPermissions(props: Props) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [currentAdminOption, setCurrentAdminOption] = useState(props.isGrafanaAdmin ? 'YES' : 'NO');
 
-  onChangeClick = () => {
-    this.setState({ isEditing: true });
+  const onChangeClick = () => {
+    setIsEditing(true);
   };
 
-  onCancelClick = () => {
-    this.setState({
-      isEditing: false,
-      currentAdminOption: this.props.isGrafanaAdmin ? 'YES' : 'NO',
-    });
+  const onCancelClick = () => {
+    setIsEditing(false);
+    setCurrentAdminOption(props.isGrafanaAdmin ? 'YES' : 'NO');
   };
 
-  onGrafanaAdminChange = () => {
-    const { currentAdminOption } = this.state;
+  const onGrafanaAdminChange = () => {
     const newIsGrafanaAdmin = currentAdminOption === 'YES' ? true : false;
-    this.props.onGrafanaAdminChange(newIsGrafanaAdmin);
+    props.onGrafanaAdminChange(newIsGrafanaAdmin);
   };
 
-  onAdminOptionSelect = (value: string) => {
-    this.setState({ currentAdminOption: value });
+  const onAdminOptionSelect = (value: string) => {
+    setCurrentAdminOption(value);
   };
 
-  render() {
-    const { isGrafanaAdmin } = this.props;
-    const { isEditing, currentAdminOption } = this.state;
-    const changeButtonContainerClass = cx('pull-right');
-    const canChangePermissions = contextSrv.hasPermission(AccessControlAction.UsersPermissionsUpdate);
+  const changeButtonContainerClass = cx('pull-right');
```

Comment:
```suggestion
```
I know this is a legacy code, but this variable is redundant and the class name can be just declared inline (see below).